### PR TITLE
Fix duplicate TOC in Sphinx docs and automate bundling of docs

### PR DESCRIPTION
**Pull request by fantavlik**
Provide easier docs bundling with tox environment and Sphinx installed as part of new 'make docs' command.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,19 +1,6 @@
 Welcome to the API reference for the Splunk SDK for Python, which describes the modules that are included in the SDK.
 For more information, see the `Splunk Developer Portal <http://dev.splunk.com/view/python-sdk/SP-CAAAEBB>`_.
 
-.. toctree::
```
Comment (fantavlik) [inline docs/index.rst:4]: This seemed to be the cause of the duplicate TOC

File: tox.ini [hunk 2]
```diff
@@ -41,3 +41,9 @@ commands =
 deps = coverage
 skip_install = true
 commands = coverage erase
+
+[testenv:docs]
+description = invoke sphinx-build to build the HTML docs
+basepython = python3.7
+deps = sphinx >= 1.7.5, < 2
```
Comment (fantavlik) [inline tox.ini:48]: By running this as a tox command we can let tox install sphinx in a Python 3 environment and skip those steps as part of docs generation

File: tox.ini [hunk 3]
```diff
@@ -26,7 +26,7 @@ application-import-names = splunk-sdk-python
 passenv = LANG
 setenv = SPLUNK_HOME=/opt/splunk
          INPUT_EXAMPLE_UPLOAD=/opt/splunk/var/log/splunk/splunkd_ui_access.log
-whitelist_externals = true
+allowlist_externals = make
```
Comment (fantavlik) [inline tox.ini:29]: This tells tox that it's OK to run `make`, see: https://tox.readthedocs.io/en/latest/example/basic.html?highlight=make#allowing-non-virtualenv-commands

**Review by zenmoto — APPROVED** [review]

**Comment by fantavlik** [comment]
I will wait to merge this until @PKing70 has a chance to look at the latest generated docs for v1.6.15 and verifies that the new script is producing sane-looking docs

**Pull request merged**